sys/net/link_layer/ieee802154/submac.c

```diff
@@ -169,6 +169,18 @@ void ieee802154_submac_rx_done_cb(ieee802154_submac_t *submac)
     }
     else {
         submac->cb->rx_done(submac);
+
+        /* The Radio HAL will be in "FB Lock" state. We need to do a state
+         * transition here in order to release it */
+        ieee802154_trx_state_t next_state = submac->state == IEEE802154_STATE_LISTEN ? IEEE802154_TRX_STATE_RX_ON : IEEE802154_TRX_STATE_TRX_OFF;
```

Comment:
Why do we go to `IEEE802154_TRX_STATE_RX_ON` in the one case and `IEEE802154_TRX_STATE_TRX_OFF` in the other?

Reply:
because the SubMAC defines a "Continuous RX" mode. If the state is `IEEE802154_STATE_LISTEN`, the radio will be always listening. Otherwise it goes to a state where it consumes less energy (TRX_OFF)